File: src/main/java/abstraction/Message.java
package abstraction;
import java.io.*;
public class Message implements Serializable {
    public Message(byte[] message, int port){
        this.message=message;
        this.port = port;
    }
    public int port;
    public byte[] message;
    public static Message restoreMessage(byte[] input){
        Object o=null;
        ByteArrayInputStream bis = new ByteArrayInputStream(input);
        ObjectInput in = null;
        try {
            in = new ObjectInputStream(bis);
            o = in.readObject();
        } catch (IOException e) {
            e.printStackTrace();
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        } finally {
            try {
                if (in != null) {
                    in.close();
                }
            } catch (IOException ex) {
                // ignore close exception
            }
        }
        return (Message) o;
    }

    public static byte[] saveMessagetoByte(Message event){
        byte[] Result=null ;
        ByteArrayOutputStream bos = new ByteArrayOutputStream();
        ObjectOutput out = null;
        try {
            out = new ObjectOutputStream(bos);
            out.writeObject(event);
            out.flush();
            Result = bos.toByteArray();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                bos.close();
            } catch (IOException ex) {
                // ignore close exception
            }
        }
        return Result;
    }
}

File: src/main/scala/abstractions/Message.scala
package abstractions

import java.io.{ByteArrayInputStream, ByteArrayOutputStream, ObjectInputStream, ObjectOutputStream}

case class Message(message: Array[Byte], port: Int)  {
  def restoreMessage(input: Array[Byte]): Message = {
    val bis = new ByteArrayInputStream(input)
    var in = new ObjectInputStream(bis)
    var o =  in.readObject
    o.asInstanceOf[Message]
  }

  def saveMessagetoByte(event: Message): Array[Byte] = {
    val bos = new ByteArrayOutputStream
    var out = new ObjectOutputStream(bos)
    out.writeObject(event)
    out.flush()
    var Result = bos.toByteArray
    bos.close()
    Result
  }

}
